### ISSUE TITLE
[Question] Porting speaker recognition from 0.11.0 to 1.0.0b

### ISSUE THREAD

**Issue by TobiasPuetzKenbun** [NVIDIA/NeMo#1613]
Hi, thanks for the great library!

I am currently porting a 0.11 speaker recognition checkpoint to 1.0.0b, I ran into some difficulties reproducing the outputs of the 0.11 checkpoint in 1.0.0b. In the 0.11 model, the decoder is `JasperDecoderForSpkrClass`, to my understanding, the corresponding class in 1.0.0b is `SpeakerDecoder`.

 In [JasperDecoderForSpkrClass.forward](https://github.com/NVIDIA/NeMo/blob/ac09a4cc2ae3fdfedaf6f6832bd2dfca94de19ec/nemo/collections/asr/jasper.py#L459), the returned embedding is the first transformation of the pooled encoder output, in the current release, [SpeakerDecoder.forward](https://github.com/NVIDIA/NeMo/blob/ede204df0d5adc54714981511b83cf6d2fa905b2/nemo/collections/asr/modules/conv_asr.py#L421) returns the last transformation of the pool. Is there a reason for this change?

**Comment by nithinraok** [comment]
Hi @TobiasPuetzKenbun ,

Did you check out [this script ](https://github.com/NVIDIA/NeMo/blob/main/scripts/asr_checkpoint_port.py)to convert your 0.11 checkpoints to 1.0? 

And to your second question yes we did that on purpose, initially when we started out speaker recognition it was first trained end to end using cross entropy loss and hence last but one linear transformed output gave good results, but when we moved loss to angular softmax, the final transformed linear output showed better results on EER (though slightly), not a big change but decided to do it. 

Let me know if you have any more questions : )

**Comment by TobiasPuetzKenbun** [comment]
Thank you for the quick answer!

> Did you check out this script to convert your 0.11 checkpoints to 1.0?

Yes, I've been using that script. It wasn't directly compatible since some module names were changed from 0.11 to 1.0, the mid1 & mid2 layers are now wrapped in a `Sequential`, so it was necessary to rename these keys in the decoder state dict to `emb_layers.0` / `emb_layers.1` before the `asr_checkpoint_port` script would work.

> And to your second question yes we did that on purpose, initially when we started out speaker recognition it was first trained end to end using cross entropy loss and hence last but one linear transformed output gave good results, but when we moved loss to angular softmax, the final transformed linear output showed better results on EER (though slightly), not a big change but decided to do it.

Thanks for the explanation, so for backward compatibility, I'll add my own legacy `SpeakerDecoder`.
